stdlib/src/main/scala/chisel3/std/tilelink/Parameters.scala

```diff
@@ -0,0 +1,58 @@
+package chisel3.std.tilelink
+
+case class TLBundleParameter(
+  a: TLChannelAParameter,
+  b: Option[TLChannelBParameter],
+  c: Option[TLChannelCParameter],
+  d: TLChannelDParameter,
+  e: Option[TLChannelEParameter]) {
+  require(
+    (b.isDefined && c.isDefined && e.isDefined) || (b.isEmpty && c.isEmpty && e.isEmpty),
+    "only AD/ABCDE channels is allowed."
+  )
+
+  def bundle(): TLBundle = new TLBundle(copy())
+
+  def isTLC = b.isDefined
+}
+
+case class TLChannelAParameter(
+  addressWidth: Int,
+  sourceWidth:  Int,
+  dataWidth:    Int,
+  sizeWidth:    Int) {
+  require(dataWidth % 8 == 0, "Width of data field must be multiples of 8")
+  def bundle(): TLChannelA = new TLChannelA(copy())
+}
+
+case class TLChannelBParameter(
+  addressWidth: Int,
+  sourceWidth:  Int,
+  dataWidth:    Int,
+  sizeWidth:    Int) {
+  require(dataWidth % 8 == 0, "Width of data field must be multiples of 8")
```

Comment:
I believe it is also required to be a power-of-2

Reply:
Yes!
See Table 4. TileLink per-link parameters.